src/main/java/org/glite/authz/pep/pip/provider/policynamespip/PolicyNamesCache.java

```diff
@@ -0,0 +1,625 @@
+
+	// Create new Entry for this path, already setting the name and the like
+	Entry entry=new Entry(path, modified);
+
+	// Parse out the value of the key into the entry, this can throw
+	// IOException or ParseException
+	parseSubjectDNvalue(entry, value);
+
+	// Now return the entry
+	return entry;
+    }
+
+    /**
+     * Parses an {@value #FILE_SFX} file for the {@link #SUBJECT_KEY} key and
+     * returns the value.
+     * @param path Path of this info file
+     * @return String with the value of the {@link #SUBJECT_KEY}
+     * @throws IOException when reading the file failed
+     */
+    private String getSubjectDNvalue(Path path) throws IOException	{
+	String name=path.toString();
+	BufferedReader reader=null;
+	StringBuilder linebuilder=new StringBuilder();
+	String newline;
+	String value=null;
+
+	try {
+	    reader=Files.newBufferedReader(path, Charset.defaultCharset());
+	} catch (IOException e)	{
+	    throw new IOException("Cannot open "+name+": "+e.getMessage());
+	}
+	
+	// initialize line
+	try {
+	    while ( (newline=reader.readLine()) != null )   {
+		// Append to existing or empty line
+		linebuilder.append(newline);
+
+		// Handle continuation char
+		int end=linebuilder.length()-1;
+		if (end>=0 && linebuilder.charAt(end)=='\\')	{
+		    linebuilder.deleteCharAt(end);
+		    continue;
+		}
+
+		// Remove leading whitespace (easiest when converting to String)
+		String line=linebuilder.toString().trim();
+		// Only look at non-empty non-comment lines
+		if (!line.isEmpty() && line.charAt(0)!='#')	{
+		    // Split into key / value and look for subjectdn
+		    int sep=line.indexOf('=');
+		    if (sep>=0 && SUBJECT_KEY.equals(line.substring(0,sep).trim())) {
+			value=line.substring(sep+1).trim();
+			break;
+		    }
+		}
+
+		// Continue with next line
+		linebuilder.setLength(0);
+	    }
+	} catch (IOException e)	{
+	    // Try to close, this might throw a new IOException. We're throwing
+	    // one in any case.
+	    reader.close();
+	    throw new IOException("Reading from "+name+" failed: "+e.getMessage());
+	}
+
+	// Close reader
+	try {
+	    reader.close();
+	} catch (IOException e)	{
+	    throw new IOException("Closing "+name+" failed: "+e.getMessage());
+	}
+
+	// All done
+	return value;
+    }
+
+    /**
+     * Parses out the different components of a subjectdn value as obtained by
+     * {@link #getSubjectDNvalue(Path)} and puts the results into the given entry.
+     * When a "file:" entry is found, it will add the correct path to the right
+     * list of dependencies.
+     * @param entry {@link Entry} to fill
+     * @param value value of the subjectdn key
+     * @throws ParseException on syntax errors in the value
+     * @throws IOException on read errors with one of the info files
+     */
+    private void parseSubjectDNvalue(Entry entry, String value)
+	throws ParseException, IOException
+    {
+	// Now parse the value part
+	int pos=0, pos2;
```

Comment:
Shorter, but a lot slower too. regexps are really slow and inefficient in java (not only in java). Even then, it's not very trivial and you get easily quite unreadable regexps. We can discuss this of course.